### PR TITLE
Notify proxy maintainers

### LINKED ISSUE #1145: Find a way to label PRs with the relevant code owners.
Currently we assign maintainers for incoming PRs with [CODEOWNERS](https://github.com/tensorflow/addons/blob/master/.github/CODEOWNERS). However, we're limited to tagging users with write access.

Seeing as a majority of our maintainers are submodule and not repository maintainers, it would be nice to devise a strategy for tagging submodule maintainers (even if it's just an @). Possibly would need to be a bot.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
Fix #1145

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
Hooray!

**Comment by gabrieldemarmiesse** [comment]
I suppose we need to remove all people who have write access, right? They already get a notification.

**Comment by seanpmorgan** [comment]
> I suppose we need to remove all people who have write access, right? They already get a notification. and also tensorflow/sig-addons-maintainers since it doesn't seem to trigger anything: [#1137 (comment)](https://github.com/tensorflow/addons/pull/1137#issuecomment-592021710)

Thanks this is fantastic!

1) Yeah we'll probably want to remove the duplicate notifications. Do you have a method in mind for doing this? Filtering members of sig-addons-maintainers?

Subsequent PR:
1) Yeah sig-addons-maintainers doesn't actually do anything. We can remove that catch all.
2) We'll need to update the CODEOWNERS to match the README docs. Then we can remove the maintainers from the README so we don't have duplicated work on PRs.

**Comment by gabrieldemarmiesse** [comment]
The easier is to hard-code in the python file the people with write access. This is not public information so I cannot fetch it automatically. I'll add a filter somewhere.

We want to minimize spamming. So the bot will only notify when the pull request is opened. Never after. It's also simpler code-wise.

File: .github/workflows/notify_codeowners.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+BLACKLIST = [
+    "tensorflow/sig-addons-maintainers",
+    "facaiy",
+    "seanpmorgan",
+    "squadrick",
+    "shun-lin",
+    "windqaq",
+    "qlzh727",
+    "guillaumekln",
+    "helinwang",
```
Comment (seanpmorgan) [inline .github/workflows/notify_codeowners.py]: Doesn't have write access. Updated the confusing codeowners in another repo

Comment (gabrieldemarmiesse) [reply .github/workflows/notify_codeowners.py]: Done. Please merge this PR before updating the codeowners. A check runs in the CI to verify that CODEOWNERS is valid.

**Pull request merged**